Tribler/Test/Core/Modules/RestApi/test_my_channel_endpoints.py

```diff
@@ -17,13 +33,32 @@ def test_my_channel_overview_endpoint_with_channel(self):
         """
         Testing whether the API returns the right JSON data if a channel overview is requested
         """
-        channel_db_handler = self.session.open_dbhandler(NTFY_CHANNELCAST)
-
         channel_json = {u'overview': {u'name': u'testname', u'description': u'testdescription',
                                       u'identifier': 'fakedispersyid'.encode('hex')}}
-        channel_db_handler._get_my_dispersy_cid = lambda: "myfakedispersyid"
-        channel_db_handler.on_channel_from_dispersy('fakedispersyid', None,
-                                                         channel_json['overview']['name'],
-                                                         channel_json['overview']['description'])
+        self.create_my_channel(channel_json[u'overview'][u'name'], channel_json[u'overview'][u'description'])
 
         return self.do_request('mychannel/overview', expected_code=200, expected_json=channel_json)
+
+    @deferred(timeout=10)
+    def test_my_channel_torrents_endpoint_no_my_channel(self):
+        """
+        Testing whether the API returns response code 404 if no channel has been created when fetching torrents
+        """
+        return self.do_request('mychannel/torrents', expected_code=404)
+
+    def verify_torrents_json(self, body):
+        torrents_dict = json.loads(body)
+        self.assertTrue(torrents_dict["torrents"])
+        self.assertEqual(len(torrents_dict["torrents"]), 1)
```

Comment:
True, the current timestamp is added to the torrent in the `Torrents` table so the content that is returned by `getTorrentsFromChannelId` is not known beforehand. That's why I've chosen for this approach.